monkestation/code/modules/clothing/shoes/magboots.dm

```diff
@@ -0,0 +1,70 @@
+/obj/item/clothing/shoes/magboots/boomboots
+
+	desc = "The ultimate in clown shoe technology."
+	name = "boom boots"
+	icon_state = "boomboot0"
+	item_state = "boomboot0"
+	magboot_state = "boomboot"
+	slowdown = SHOES_SLOWDOWN+1
+	item_color = "boomboots"
+	actions_types = list(/datum/action/item_action/toggle)
+	pocket_storage_component_path = /datum/component/storage/concrete/pockets/shoes/clown
+	var/datum/component/waddle
+	var/enabled_waddle = TRUE
+
+/obj/item/clothing/shoes/magboots/boomboots/Initialize()
+	. = ..()
+	AddComponent(/datum/component/squeak, list('monkestation/sound/misc/Boomboot1.ogg'=1), 50)
+
+/obj/item/clothing/shoes/magboots/boomboots/equipped(mob/user, slot)
+	. = ..()
+	if(slot == ITEM_SLOT_FEET)
+		if(enabled_waddle)
+			waddle = user.AddComponent(/datum/component/waddling)
+		if(user.mind && user.mind.assigned_role == "Clown")
+			SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "clownshoes", /datum/mood_event/clownshoes)
+
+/obj/item/clothing/shoes/magboots/boomboots/item_action_slot_check(slot, mob/user)
+	if(slot == ITEM_SLOT_FEET)
+		return 1
+
+/obj/item/clothing/shoes/magboots/boomboots/attack_hand(mob/user)
+	if(iscarbon(user))
+		var/mob/living/carbon/C = user
+		if(src == C.shoes && magpulse)
+			to_chat(user, "<span class='userdanger'>The boomboots anti-tamper system doesn't allow you to remove them while on!</span>")
```

Comment:
Unequipping them by dragging to hand still works. Not sure if that can be fixed while still allowing for others to attempt to take them